release/models/acl/openconfig-icmpv4-types.yang

```diff
@@ -0,0 +1,530 @@
+module openconfig-icmpv4-types {
+
+  yang-version "1";
+  namespace "http://openconfig.net/yang/openconfig-icmpv4-types";
+
+  prefix "oc-icmpv4-types";
+
+  import openconfig-extensions { prefix oc-ext; }
+
+  organization "OpenConfig working group";
+
+  contact
+    "OpenConfig working group
+    www.openconfig.net";
+
+  description
+    "OpenConfig module defining the types and coresponding codes for
+    ICMPv4.";
+
+  oc-ext:openconfig-version "0.1.0";
+
+  revision "2023-01-26" {
+    description
+      "Initial revision of ICMPv4 types module.";
+    reference "0.1.0";
+  }
+
+  identity TYPE {
+    description
+      "Base identity for ICMPv4 codes";
+  }
+
+  identity CODE {
+    description
+      "Base identity for ICMPv4 codes.";
+  }
+
+  identity ECHO_REPLY {
+    description
+      "ICMP echo reply, value 0.";
+    base TYPE;
+    reference "RFC792: INTERNET CONTROL MESSAGE PROTOCOL";
+  }
+
+  identity DST_UNREACHABLE {
+    description
+      "ICMP destination unreachable, value 3.";
+    base TYPE;
+    reference "RFC792: INTERNET CONTROL MESSAGE PROTOCOL";
+  }
+
+  identity REDIRECT {
+    description
+      "ICMP redirect, value 5.";
+    base TYPE;
+    reference "RFC792: INTERNET CONTROL MESSAGE PROTOCOL";
+  }
+
+  identity ECHO {
+    description
+      "ICMP echo, value 8.";
+    base TYPE;
+    reference "RFC792: INTERNET CONTROL MESSAGE PROTOCOL";
+  }
+
+  identity ROUTER_ADVERTISEMENT {
+    description
+      "ICMP router advertisement, value 9.";
+    base TYPE;
+    reference "RFC1256: ICMP Router Discovery Messages";
+  }
+
+  identity ROUTER_SELECTION {
+    description
+      "ICMP router selection, value 10.";
+    base TYPE;
+    reference "RFC1256: ICMP Router Discovery Messages";
+  }
+
+  identity TIME_EXCEEDED {
+    description
+      "ICMP TTL exceede, value 11.";
+    base TYPE;
+    reference "RFC792: INTERNET CONTROL MESSAGE PROTOCOL";
+  }
+
+  identity PARAM_PROBLEM {
+    description
+      "ICMP parameter problem, value 12.";
+    base TYPE;
+    reference "RFC792: INTERNET CONTROL MESSAGE PROTOCOL";
+  }
+
+  identity TIMESTAMP {
+    description
+      "ICMP timestamp, value 13.";
+    base TYPE;
+    reference "RFC792: INTERNET CONTROL MESSAGE PROTOCOL";
+  }
+
+  identity TIMESTAMP_REPLY {
+    description
+      "ICMP timestamp reply, value 14.";
+    base TYPE;
+    reference "RFC792: INTERNET CONTROL MESSAGE PROTOCOL";
+  }
+
```

Comment:
I have added TRACEROUTE, even if it is deprecated, it seems it is still used a bit. I highlighted the deprecated in the description.

Reply:
Thanks, accepted the suggestion :-)